GPy/testing/ccd_test_1.py

```diff
@@ -0,0 +1,132 @@
+        self.ndims = self.m.num_params
+        self.true_mode = self.m.peak_loc
+        self.stepsize = 0.2
+
+    def tearDown(self):
+        # model could be memory intensive- so remove after test!
+        self.m = None
+
+    def test_ccd_placement(self):
+        # m2 = SimpleModel('simple', 2)
+        self.m.optimize()
+
+        # assert np.all(np.isclose(self.m.numerical_parameter_hessian(), np.array([[2,0],[0,2]]))), "Numerical approximation to Hessian doesn't match."
+        assert np.all(np.isclose(self.m.numerical_parameter_hessian(), np.eye(self.ndims)*2)), "Numerical approximation to Hessian doesn't match. Error in numerical_parameter_hessian()."
+
+        #check the optimizing step-found the maximum correctly.
+        for i in range(self.m.num_params):
+            assert np.isclose(self.m.params[i], i+1, atol=0.01), "Failed to find likelihood maximum of test model's parameter while testing CCD"
+        # get the CCD positions and log likelihoods at those locations.
+        ccdpos, ccdres, scalings, z = self.m.CCD()
+        # (optimise again as CCD moves the parameter values)
+        self.m.optimize()
+        # calculate euclidean distance of ccdpoints except centralpoint from true mode.
+        dists = np.sum((ccdpos - self.true_mode) ** 2, 1)[1:]
+        dists - np.mean(dists)
+        # basically checks that all CCD points are equidistant from the central point.
+        assert np.all(np.isclose(dists, np.mean(dists),
+                                 atol=0.01)), "CCD placement error - Points should be equidistant!"
+        assert np.all(np.isclose(np.sum(ccdres[1:] / ccdres[0]), 4.7619,
+                                 atol=0.1)), "CCD placement error - off-centre locations should have log likelihood" \
+                                             " ratios to central point summing to 4.76 times the centre, for nd" \
+                                             " symmetrical Quadratic test case"
+
+    @nottest
+    def find_likes(self, model, stepsize=0.3,rangemin=-2,rangemax=7):
+        params = model.parameter_names_flat()
+        param_ranges = []
+        for param in params:
+            param_ranges.append(np.arange(rangemin, rangemax, stepsize))
+        combs = itertools.product(*param_ranges)
+        llsum = 0
+        for el in combs:
+            llsum += np.exp(-model._objective(el))
+        return llsum
+
+    def test_ccd_integration(self, model=None):
+        if model is None:
+            model = self.m
+        ls = self.find_likes(self.m, self.stepsize)
+        numsum = ls*(self.stepsize**self.ndims)
+        model.optimize()
+        hes = model.numerical_parameter_hessian()
+        # Laplace approximation to find marginal likelihoods of submodels, integrate out hyperparams of kernel matrix and likelihood params
+        hessum = np.exp(model.log_likelihood()) * 1 / np.sqrt(np.linalg.det(1 / (2 * np.pi) * hes))
+        print hessum, hes
+        print numsum
```

Comment:
This print statements are causing python 3 compatibility issues. I think ideally we avoid printing during tests though if possible perhaps it is better to remove them.